.github/workflows/ci.yml

```diff
@@ -0,0 +1,44 @@
+name: core2_firmware CI
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  clang-format-check:
+    name: Clang-format formatting check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Run clang-format style check.
+      uses: jidicula/clang-format-action@v4.11.0
+      with:
+        clang-format-version: '14'
+        include-regex: ^.*\.(cpp|hpp)$
+
+  build:
+    name: Build the project
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+      - name: Prepare environment
+        run: |
+          git submodule update --init
```

Comment:
Instead of running git command directly, add `submodules` options to checkout action

Reply:
Done.